src/main/java/dev/task/dndquest/controller/PlayCharacterController.java

```diff
@@ -23,15 +17,9 @@ public class PlayCharacterController {
     PlayCharacterService playCharacterService;
     DtoMapper<PlayCharacter, PlayCharacterResponseDto, PlayCharacterRequestDto> mapper;
 
-    @ResponseStatus(code = HttpStatus.CREATED)
     @PostMapping
+    @ResponseStatus(code = HttpStatus.CREATED)
     public void createCharacter(@Valid @RequestBody PlayCharacterRequestDto playCharacterDto){
-        try {
-            PlayCharacter playCharacter = mapper.mapToEntity(playCharacterDto);
-            playCharacterService.save(playCharacter);
-        } catch (DBException e) {
-            throw new ResponseStatusException(
-                    HttpStatus.PRECONDITION_FAILED, e.getMessage(), e);
-        }
+        playCharacterService.save(mapper.mapToEntity(playCharacterDto));
```

Comment:
The controller doesn't have to know about the mapper. Why not do that on a service level?
Try to keep controllers as thin as possible

Reply:
understood